jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtil.java

```diff
@@ -727,6 +725,27 @@ private static void ensureHasPort(Container container, ContainerPort port) {
         ports.add(port);
     }
 
+    private static boolean isSimpleFieldType(Class<?> type) {
+        ArrayList<Class<?>> simpleFieldTypes = new ArrayList<>();
+        simpleFieldTypes.add(String.class);
+        simpleFieldTypes.add(Double.class);
+        simpleFieldTypes.add(Float.class);
+        simpleFieldTypes.add(Long.class);
+        simpleFieldTypes.add(Integer.class);
+        simpleFieldTypes.add(Short.class);
+        simpleFieldTypes.add(Character.class);
+        simpleFieldTypes.add(Byte.class);
+        simpleFieldTypes.add(double.class);
+        simpleFieldTypes.add(float.class);
+        simpleFieldTypes.add(long.class);
+        simpleFieldTypes.add(int.class);
+        simpleFieldTypes.add(short.class);
+        simpleFieldTypes.add(char.class);
+        simpleFieldTypes.add(byte.class);
```

Comment:
There are several things I don't understand here.
- Why did we replace the `Set` with an `ArrayList`?
- Why are we no longer statically initilizing the collection (The collection is now created each time we invoke the method)?
- If this collection is to be different depending on where it is invoked (e.g. DefaultControllerEnricher), why aren't we creating an instantiable dedicated util class where the Collection can be customized?

Reply:
This collection only seems to be getting used to distinguish whether a type is primitive or not. I don't think it's contents would vary depending upon the caller. I wanted to use it in RouteEnricher but I found out it's being initialized in DefaultControllerEnricher; which seemed weird to me.

Reply:
regarding replacing `Set` with `ArrayList` I think it's a mistake. Let me revert it to `Set`.

Reply:
:+1: 
I'm still missing this _"Why are we no longer statically initializing the collection (The collection is now created each time we invoke the method)?"_

Reply:
ohk, let me revert to static initialization inside KubernetesResourceUtil

Reply:
I think this one still pending.